### PR TITLE
adding tests for json deserialization and use System.Text.Json

### PR DESCRIPTION
**Pull request by colombod**

### CODE REVIEW

File: src/lobe.TestApp/Program.cs [hunk 1]
```diff
@@ -11,33 +16,82 @@ class Program
     {
         static void Main(string[] args)
         {
-            var signatureFilePath = args[0];
-            var imageToClassify = args[1];
+            var parser = CreateParser();
 
-            ImageClassifier.Register("onnx", () => new OnnxImageClassifier());
+            parser.Invoke(args);
```
Comment (adiazulay) [inline src/lobe.TestApp/Program.cs:21]: I'm not super familiar with C#, but is it a common pattern to run everything inside of the parser?

File: src/lobe.TestApp/Program.cs [hunk 2]
```diff
@@ -11,33 +16,82 @@ class Program
...
-                foreach (var file in files)
+            var predictionEndpointOption = new Option<Uri>("--prediction-endpoint");
+
+            rootCommand.AddOption(signatureFileOption);
+            rootCommand.AddOption(imageFileOption);
+            rootCommand.AddOption(imageFolderOption);
+            rootCommand.AddOption(predictionEndpointOption);
+
+            rootCommand.Handler = CommandHandler.Create<FileInfo, FileInfo, DirectoryInfo, Uri>(
+                (signatureFile, imageFile, imageFolder, predictionEndpoint) =>
```
Comment (adiazulay) [inline src/lobe.TestApp/Program.cs:43]: will this work if one of the args is missing?

File: src/lobe.TestApp/Program.cs (hunk 1 above)
Comment (colombod) [reply src/lobe.TestApp/Program.cs:21]: The command line library defines parsers and you can create handlers for commands. This is a common pattern with that library when creating cli tools

File: src/lobe.TestApp/Program.cs (hunk 2 above)
Comment (colombod) [reply src/lobe.TestApp/Program.cs:43]: I should add check that endpoint of signature must be provided and same that either an i mange or a folder (and file system entries must exist)

**Pull request merged**